rclpy/src/rclpy/context.hpp

```diff
@@ -0,0 +1,59 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLPY__CONTEXT_HPP_
+#define RCLPY__CONTEXT_HPP_
+
+#include <Python.h>
+#include <pybind11/pybind11.h>
+#include <rcl/context.h>
+#include <rcl/types.h>
+
+#include <stdexcept>
+
+extern "C" {
+#include "rclpy_common/handle.h"
+}
```

Comment:
This is out-of-scope for this review, but is there any reason not to incorporate these linkage directives in the header itself? That seems to be the more common practice, but I could be missing something.

Reply:
Laziness - I didn't need to type the `#ifndef __cplusplus` this way. It looks like `handle.h` already has it though, so removed this in 83b9368bc3ceaec955f1fe2c7635da41e09ff2d4